lib/accumulate_distribute/meta/init_state.js

```diff
@@ -22,9 +23,9 @@ const initState = (args = {}) => {
   const orderAmounts = genOrderAmounts.gen(args)
 
   let totalAmount = 0
-  orderAmounts.forEach(a => { totalAmount += a })
+  orderAmounts.forEach(a => { totalAmount = nBN(totalAmount).plus(a).toNumber() })
 
-  if (Math.abs(totalAmount - amount) > DUST) {
+  if (nBN(totalAmount).minus(amount).abs().toNumber() > DUST) {
```

Comment:
abs() can introduce issues here:

lets say you have a remainder of 0.1
and you go to -0.1 next round
and dust is 0.05
then it does another round